app/test/features/chat_ng/widgets/chat_messages_list_test.dart

```diff
@@ -0,0 +1,406 @@
+import 'package:acter/features/chat/providers/chat_providers.dart' as chat;
+import 'package:acter/features/chat_ng/models/chat_room_state/chat_room_state.dart';
+import 'package:acter/features/chat_ng/providers/chat_room_messages_provider.dart';
+import 'package:acter/features/chat_ng/providers/notifiers/chat_room_messages_notifier.dart';
+import 'package:acter/features/chat_ng/widgets/chat_messages.dart';
+import 'package:acter/features/chat_ng/widgets/events/chat_event.dart';
+import 'package:acter/features/labs/model/labs_features.dart';
+import 'package:acter/features/labs/providers/labs_providers.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+// import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mocktail/mocktail.dart';
+
+import '../../../helpers/test_util.dart';
+
+class MockTimelineStream extends Mock implements TimelineStream {
+  @override
+  Future<bool> paginateBackwards(int count) async => false;
+}
+
+class MockRef extends Mock implements Ref {}
+
+class MockChatRoomMessagesNotifier extends Mock
+    implements ChatRoomMessagesNotifier {
+  final ChatRoomState initalState;
+
+  MockChatRoomMessagesNotifier(this.initalState);
+
+  @override
+  ChatRoomState get state => initalState;
+
+  @override
+  void Function() addListener(
+    void Function(ChatRoomState) listener, {
+    bool fireImmediately = true,
+  }) {
+    if (fireImmediately) {
+      listener(state);
+    }
+    return () {};
+  }
+
+  @override
+  Future<void> loadMore({bool failOnError = true}) async {}
+}
+
+void main() {
+  group('ChatMessages Widget Tests', () {
+    final testRoomId = 'test-room-id';
+
+    setUp(() {
+      registerFallbackValue(const ChatRoomState());
+      registerFallbackValue(true); // Register fallback for boolean parameters
+      registerFallbackValue(false); // Register fallback for boolean parameters
```

Comment:
I assume this comes from some AI generation code, but are these really _both_ necessary?